powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/OperationServiceBehavior.java

```diff
@@ -813,6 +821,16 @@ private ProximityCheckResult fetchProximityCheckResult(final OperationEntity ope
         }
     }
 
+    private static void parseDeviceFromUserAgent(Map<String, Object> additionalData) {
```

Comment:
I have objections to the method name, even if it is _private_. `parse` evokes a return value to me, which is missing. In general, I do not like side effects, mainly hidden. This method is _parse and add_.